### PR TITLE
Task/black button synchronous ops

### LINKED ISSUE #37: Implement Integration of the Synchronous part of the Black Button protocol
- [x] Synchronous mode: implement S
- [x] Synchronous mode: implement request holding mechanism to wait for the answer
- [x] Synchronous mode: implement commands in the client
- [ ] Synchronous mode: integrate with real Context Broker

### PR DESCRIPTION
**Pull request by dmoranj**
Fixes #37

### CODE REVIEW

File: test/unit/contextAvailabilityResponses/registerDeviceSuccess.json [hunk 1]
```diff
@@ -0,0 +1,4 @@
+{
+  "duration" : "P1M",
+  "registrationId" : "6319a7f5254b05844116584d"
+}
```
Comment (gtorodelvalle) [inline test/unit/contextAvailabilityResponses/registerDeviceSuccess.json]: Final blank line appreciated... ;)

**Comment by gtorodelvalle** [comment]
Looking great to me apart from just one minor comment... :)

File: test/unit/contextAvailabilityResponses/registerDeviceSuccess.json (hunk 1 above)
Comment (dmoranj) [inline test/unit/contextAvailabilityResponses/registerDeviceSuccess.json]: Done

**Comment by gtorodelvalle** [comment]
LGTM! :) Thanks @dmoranj ;)

**Pull request merged**